unit-tests-java8/src/test/java/org/eclipse/collections/test/RichIterableTestCase.java

```diff
@@ -272,6 +272,32 @@ default void RichIterable_getLast()
     }
 
     @Test
+    default void RichIterable_getOnly()
+    {
+        RichIterable<Integer> iterable = this.newWith(3);
+        Integer only = iterable.getOnly();
+        assertThat(only, is(3));
+
+        Iterator<Integer> iterator = iterable.iterator();
+        assertThat(iterator.next(), is(only));
+        assertThat(iterator.hasNext(), is(false));
+    }
+
+    @Test
+    default void RichIterable_getOnly_exception_when_empty()
+    {
+        RichIterable<Object> emptyIterable = this.newWith();
+        assertThrows(IllegalStateException.class, () -> emptyIterable.getOnly());
+    }
+
+    @Test
+    default void RichIterable_getOnly_exception_when_two_items()
+    {
+        RichIterable<Object> iterable = this.newWith(1, 2);
+        assertThrows(IllegalStateException.class, () -> iterable.getOnly());
+    }
```

Comment:
Thank you for your translation!

> The suggestion is to combine the 3 methods into a single method

I did not override each methods. So I combine 3 methods into a single method.

> add `this.newWith(1, 1).getOnly()`

Should I implement the test method like following?

1. I add a test method (named like `getOnly_when_multiple_same_values()`) at `RichIterableTestCase` and `AbstractRichIterableTestCase`.  
This test method verifies that  `getOnly()` throws `IllegalStateException`.
1. Run each tests then some classes will be failed (`UnifiedSetTest`, `HashBagTest` etc...).
1. I override each failed test methods. Then overriden methods verify that `getOnly()` must not throw `IllegalStateException`.